x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectBuilderTests.java

```diff
@@ -126,4 +134,19 @@ public void testBuildAutodetectCommand_givenPersistModelState() {
     private AutodetectBuilder autodetectBuilder(Job job) {
         return new AutodetectBuilder(job, filesToDelete, logger, env, settings, nativeController, processPipes);
     }
+
+    public void testBuildAutodetect() throws Exception {
+        Job.Builder job = buildJobBuilder("unit-test-job");
+
+        autodetectBuilder(job.build()).build();
+
+        assertThat(filesToDelete, hasSize(3));
+
+        verify(nativeController).startProcess(commandCaptor.capture());
+        verifyNoMoreInteractions(nativeController);
+
+        List<String> command = commandCaptor.getValue();
+        String commandString = command.toString();
+        assertTrue(commandString.contains(" --config="));
```

Comment:
The current assertion is relying on the way Java chooses to print a `List`.  This is more precise:

```suggestion
        assertThat(command, hasItem(matchesPattern("--config=.*\\.json")));
```

Plus you'll need extra imports to go with that.  IntelliJ should add them automatically but if not then add:

```
import static org.hamcrest.Matchers.hasItem;
import static org.hamcrest.Matchers.matchesPattern;
```